nova_vm/src/ecmascript/builtins/promise/data.rs

```diff
@@ -8,23 +8,36 @@ use crate::{ecmascript::{builtins::control_abstraction_objects::promise_objects:
 pub struct PromiseHeapData {
     pub(crate) object_index: Option<OrdinaryObject>,
     pub(crate) promise_state: PromiseState,
-    pub(crate) promise_fulfill_reactions: Option<PromiseReactions>,
-    pub(crate) promise_reject_reactions: Option<PromiseReactions>,
-    pub(crate) promise_is_handled: bool,
 }
 
-#[derive(Debug, Clone, Copy, Default)]
+#[derive(Debug, Clone)]
 pub(crate) enum PromiseState {
-    #[default]
-    Pending,
+    Pending {
+        fulfill_reactions: Option<PromiseReactions>,
+        reject_reactions: Option<PromiseReactions>,
+        /// True if the resolution state of this promise depends on another
+        /// promise or thenable that hasn't fulfilled or rejected yet.
+        is_resolved: bool,
+    },
     Fulfilled {
         promise_result: Value,
     },
     Rejected {
         promise_result: Value,
+        is_handled: bool,
     },
 }
 
+impl Default for PromiseState {
```

Comment:
**praise:** Good idea

Reply:
There was already a `#[derive(Default)]` for `PromiseState`, but that could no longer be derived because `PromiseReactions` doesn't have a default. So I had to manually implement it instead.